pkg/apis/cluster/v1alpha2/machine_types.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+)
+
+const (
+	// MachineFinalizer is set on PrepareForCreate callback.
+	MachineFinalizer = "machine.cluster.sigs.k8s.io"
+
+	// MachineClusterLabelName is the label set on machines linked to a cluster.
+	MachineClusterLabelName = "cluster.sigs.k8s.io/cluster-name"
+
+	// MachineControlPlaneLabelName is the label set on machines part of a control plane.
+	MachineControlPlaneLabelName = "cluster.sigs.k8s.io/control-plane"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+/// [Machine]
+// Machine is the Schema for the machines API
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=machines,shortName=ma
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="ProviderID",type="string",JSONPath=".spec.providerID",description="Provider ID"
+// +kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase",description="Machine status such as Terminating/Pending/Running/Failed etc"
+// +kubebuilder:printcolumn:name="NodeName",type="string",JSONPath=".status.nodeRef.name",description="Node name associated with this machine",priority=1
+type Machine struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MachineSpec   `json:"spec,omitempty"`
+	Status MachineStatus `json:"status,omitempty"`
+}
+
+/// [Machine]
+
+/// [MachineSpec]
+// MachineSpec defines the desired state of Machine
+type MachineSpec struct {
+	// ObjectMeta will autopopulate the Node created. Use this to
```

Comment:
Do we actually expect this to affect the created Node? That seems more like a concern for the bootstrap controller than something explicitly on the MachineSpec.

I'm not sure I'm completely convinced here that we need an embedded ObjectMeta on the MachineSpec type.

Reply:
Ah, I see why this was confusing now...  ObjectMeta should not be needed on the MachineSpec, only on the MachineTemplateSpec embedded type:

See [PodTemplateSpec](https://github.com/kubernetes/kubernetes/blob/3178e69b3920f5a11ef4902a89c30df49440c9d8/staging/src/k8s.io/api/core/v1/types.go#L3267) and [PodSpec](https://github.com/kubernetes/kubernetes/blob/3178e69b3920f5a11ef4902a89c30df49440c9d8/staging/src/k8s.io/api/core/v1/types.go#L2816) for comparison